app/vmalert/datasource/client.go

```diff
@@ -56,22 +65,17 @@ type keyValue struct {
 	value string
 }
 
-// Clone makes clone of VMStorage, shares http client.
-func (s *VMStorage) Clone() *VMStorage {
-	ns := &VMStorage{
+// Clone clones shared http client and other configuration to the new client.
+func (s *Client) Clone() *Client {
+	ns := &Client{
 		c:                s.c,
 		authCfg:          s.authCfg,
 		datasourceURL:    s.datasourceURL,
 		appendTypePrefix: s.appendTypePrefix,
 		queryStep:        s.queryStep,
 
-		dataSourceType:     s.dataSourceType,
-		evaluationInterval: s.evaluationInterval,
```

Comment:
Those fields don't need to be inherited, since we should always pass the new values to override them. 
Removed from this pr, will create a separate one.